app/services/samples/metadata/update_service.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Samples
+  module Metadata
+    # Service used to Update Samples::Metadata
+    class UpdateService < BaseService
+      SampleMetadataUpdateError = Class.new(StandardError)
+      attr_accessor :sample, :metadata, :analysis_id
+
+      def initialize(project, sample, user = nil, params = {})
+        super(user, params)
+        @project = project
+        @sample = sample
+        @metadata = params['metadata']
+        @analysis_id = params['analysis_id']
+      end
+
+      def execute # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength
+        authorize! sample.project, to: :update_sample?
+        metadata_fields_update_status = { updated: [], not_updated: [] }
+
+        if @project.id != @sample.project.id
+          raise SampleMetadataUpdateError,
+                I18n.t('services.samples.metadata.sample_does_not_belong_to_project', sample_name: @sample.name,
+                                                                                      project_name: @project.name)
+        end
+
+        if @metadata.nil? || @metadata == {}
```

Comment:
**nitpick (blocking):** can you move the first part of this if statement into `validate_metadata_param`. After doing that you won't need the if/else logic since the if statement would be in the `validate_metadata_param`.

Reply:
Fixed in b78174939f9a80456d3eb8fbc8342885608f718e